common/src/chain/config/regtest.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    convert::Infallible,
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+
+use crate::{
+    chain::{
+        block::timestamp::BlockTimestamp,
+        config::{decode_hex, StakePoolData, DEFAULT_INITIAL_MINT, MIN_STAKE_POOL_PLEDGE},
+        tokens::OutputValue,
+        transaction::Destination,
+        Genesis, PoolId, TxOutput,
+    },
+    primitives::{per_thousand::PerThousand, Amount},
+};
+use crypto::{
+    key::{PrivateKey, PublicKey},
+    vrf::{VRFPrivateKey, VRFPublicKey},
+};
+
+#[derive(Clone, Debug)]
+pub struct GenesisStakingSettings {
+    pool_id: PoolId,
+    stake_private_key: PrivateKey,
+    vrf_private_key: VRFPrivateKey,
+}
+
+impl GenesisStakingSettings {
+    pub fn new(settings: &str) -> Self {
+        let settings_parts: Vec<&str> = settings.split(',').map(|s| s.trim()).collect();
+
+        let pool_id = settings_parts
+            .iter()
+            .find_map(|s| s.strip_prefix("genesis_pool_id:"))
+            .or(Some(
+                "123c4c600097c513e088b9be62069f0c74c7671c523c8e3469a1c3f14b7ea2c4",
+            ))
+            .map(decode_hex::<PoolId>)
+            .expect("Pool Id decoded");
+
+        let stake_private_key = settings_parts
+            .iter()
+            .find_map(|s| s.strip_prefix("genesis_stake_private_key:"))
+            .or(Some(
+                "008717e6946febd3a33ccdc3f3a27629ec80c33461c33a0fc56b4836fcedd26638",
+            ))
+            .map(decode_hex::<PrivateKey>)
+            .expect("Private key decoded");
+
+        let vrf_private_key = settings_parts
+            .iter()
+            .find_map(|s| s.strip_prefix("genesis_vrf_private_key:"))
+            .or(Some(
+                "003fcf7b813bec2a293f574b842988895278b396dd72471de2583b242097a59f06e9f3cd7b78d45750afd17292031373fddb5e7a8090db51221038f5e05f29998e",
+            ))
+            .map(decode_hex::<VRFPrivateKey>)
+            .expect("VRF private key decoded");
```

Comment:
I am a bit conflicted about the `.expect`s here. I know this is supposed to be used for testing only. However, it is still being parsed from user input from the command line, so some sort of nicer error reporting is probably preferable.

A related question is what to do about unrecognised keys. Let's say the user makes a typo `genesis_vrf_pirvate_key:BLAH`, then vrf key is silently going to be set to the default value instead of `BLAH`. That may be quite annoying to diagnose.

Reply:
> I am a bit conflicted about the .expects here

Yep agreed. I've already removed these locally.

Reply:
This was a tricky one...

`Display` is required for `Clap` for custom types but it doesn't really elaborate why. I thought it was just using this for user `--help` etc so I just did a `format!("{:?}...)`, but then everything broke. It turned out that when not provided on the command line, it uses `Display` to stringify `default()`, and then calls from_str() later on to parse the value.

So when passed on the command line, `GenesisStakingSettings::from_str(settings: &str)` would have `settings` set to whatever was taken from the command line, however when not passed on the command line, `settings` contained something like `{pool_id: Id<Pool>{0x123c4c600097c513e088b9be62069f0c74c7671c523c8e3469a1c3f14b7ea2c4}, stake_private_key: PrivateKey { key: Secp256k1Schnorr(Secp256k1PrivateKey { data: SecretKey(#0fd5243c90a0082f) }) },...`, all because my `format!()` in the `impl Display` wasn't "round tripping" the value as stated in the `Clap` documentation.

All good, it's working now.